nailgun/entities.py

```diff
@@ -1359,11 +1375,35 @@ def __init__(self, server_config=None, **kwargs):
                 required=True,
             ),
         }
+        super(HostClasses, self).__init__(server_config, **kwargs)
         self._meta = {
-            'api_path': 'api/v2/hosts/:host_id/puppetclass_ids',
             'server_modes': ('sat'),
+            'api_path': '{0}/puppetclass_ids'.format(
+                # pylint:disable=no-member
+                self.host.path('self')
+            )
         }
-        super(HostClasses, self).__init__(server_config, **kwargs)
+
+    def read(
+            self,
+            entity=None,
+            attrs=None,
+            ignore=('host', 'puppetclass', 'id',)
+    ):
+        """:class:`HostClasses` requires that ``host`` attribute be provided
+        Also passing ``puppetclass`` attribute as we need to compensate for the
+        unusual format of responses from the server.
+        Ignoring ``id`` attribute as current entity can't be accessed using
+        unique identification value
+
+        """
+        if entity is None:
+            entity = type(self)(
+                self._server_config,
+                host=self.host,  # pylint:disable=E1101
+                puppetclass=self.puppetclass,  # pylint:disable=E1101
+            )
+        return super(HostClasses, self).read(entity, attrs, ignore)
```

Comment:
puppetclass_id = attrs.pop('puppetclass_id')
        if puppetclass_id is None:
            attrs['puppetclass'] = None
        else:
            attrs['puppetclass'] = {'id': puppetclass_id}